corehq/apps/app_manager/fields.py

```diff
@@ -346,6 +346,16 @@ def _get_app_type_choices(self, as_dict=True):
             choices = map(lambda c: c._asdict(), choices)
         return choices
 
+    def _get_app_type_choices_for_cases(self, has_unknown_case_types=False):
```

Comment:
Before unknown option would only show up for cases if you had deleted forms or remote app forms. needless to say that was wrong and was why icds couldn't see them (and why the modal sometimes worked)